src/test/java/com/solvd/mobiletesting/GiftCardDeliveryTest.java

```diff
@@ -0,0 +1,136 @@
+package com.solvd.mobiletesting;
+
+import com.qaprosoft.carina.core.foundation.IAbstractTest;
+import com.solvd.mobiletesting.android.ChromeApp;
+import com.solvd.mobiletesting.android.page.HomePage;
+import com.solvd.mobiletesting.base.element.LocationPopUpBase;
+import com.solvd.mobiletesting.base.element.MenuBarBase;
+import com.solvd.mobiletesting.base.element.TopDropMenuBase;
+import com.solvd.mobiletesting.base.page.*;
+import com.solvd.mobiletesting.base.utils.*;
+import com.zebrunner.carina.utils.R;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+import java.util.List;
+
+public class GiftCardDeliveryTest implements IAbstractTest {
+
+    @Test(testName = "verify that delivery is impossible to Belarus")
+    public void verifyDeliveryImpossibleLocationMessageTest() {
+        HomePageBase homePage = initPage(getDriver(), HomePageBase.class);
+        homePage.open();
+        LocationPopUpBase locationPopUp = homePage.getLocationPopUp();
+        locationPopUp.clickNotChangeButton();
+        String actualLocation = homePage.getActualDeliveryLocation();
+        Assert.assertEquals(actualLocation, R.TESTDATA.get("impossibleDelivery"), "The location is not " + R.TESTDATA.get("impossibleDelivery"));
+        MenuBarBase menuBar = homePage.getMenuBar();
+        GiftCardPageBase giftCardPage = menuBar.clickGiftCardTabButton();
+        giftCardPage.clickOccasionButton();
+        MobileContextUtils contextHelper = new MobileContextUtils();
+        contextHelper.switchMobileContext(MobileContextUtils.View.NATIVE);
+        ChromeApp chromeApp = new ChromeApp(getDriver());
+        chromeApp.clickSwitchTabButton();
+        chromeApp.createNewTab();
+        contextHelper.switchMobileContext(MobileContextUtils.View.WEB);
+        ResultPageBase resultPage = giftCardPage.clickTypeOfCard("MAIL");
+        List<String> messages = resultPage.getValidationMessages();
+
+        SoftAssert softAssert = new SoftAssert();
```

Comment:
Marina mentioned it in my PR that it is not advisable to use Asserts and SoftAsserts in one test. Please consider if it's applicable for yours.

Reply:
 In these cases I check something depended on location and tests have a lot of results, but if location is wrong all results will be failed, so location   is verified before getting test results